### DISCUSSION TITLE
LabelPrimitive cause unexpected error.

### DISCUSSION THREAD

**Discussion by t-rosa** [radix-ui/primitives#2020] [Help] [answered]
Hello!

Whenever I use the LabelPrimitive, this one cause unexpected error :

![image](https://user-images.githubusercontent.com/56064543/224746590-00a9b037-9eed-4dc6-896b-52672601de40.png)

I use vite, react-router and react-hook-form, this work fine with html base label but impossible with the primitive.

Any idea ? 

(PS: Same issue with Select primitives, but it concern a null 'useMemo' instead)

**Comment by benoitgrelard** [comment, marked as answer]
That looks completely unrelated to Radix but rather a setup issue. This is essentially telling you that `React` is `null`, so it looks like it's not resolving the React dependency correctly.

**Reply by t-rosa** [reply]
This is weird because other radix primitive like scroll-area works fine and this really happen only when I use radix Label.Root component. I understand that React is undefined, but how ? And why this happen only when I use Label.Root ? This is the custom component I use :
![image](https://user-images.githubusercontent.com/56064543/224924822-87abe72f-383d-48de-866c-59db08dbb18f.png)
Same error appear with the official example from the docs. To be sure, I import React everywhere and I rerun packages installation, but still didn't work.

**Reply by t-rosa** [reply]
> This is weird because other radix primitive like scroll-area works fine and this really happen only when I use radix Label.Root component. I understand that React is undefined, but how ? And why this happen only when I use Label.Root ? This is the custom component I use : ![image](https://user-images.githubusercontent.com/56064543/224924822-87abe72f-383d-48de-866c-59db08dbb18f.png) Same error appear with the official example from the docs. To be sure, I import React everywhere and I rerun packages installation, but still didn't work.

Ok I found the problem, I'm using a monorepo setup and dependencies wasn't install at the right place, totally my bad.
